vars/uploadArtifacts.groovy

```diff
@@ -4,23 +4,29 @@ void call(Map args = [:]) {
     def manifestFilename = args.manifest ?: 'builds/manifest.yml'
     def buildManifest = lib.jenkins.BuildManifest.new(readYaml(file: manifestFilename))
 
-    def artifactPath = buildManifest.getArtifactRoot("${JOB_NAME}", "${BUILD_NUMBER}")
+    def buildNumber = args.build_number ?: "${BUILD_NUMBER}"
```

Comment:
Aren't those always present in all jobs? If they are not, we should either add them everywhere or remove them from everywhere. I would make sure the test workflow has them and restore the old code here that assumes they exist.

Reply:
build_number for test is the build on which we want to run tests on. That's why took it as a parameter to the job.

Reply:
You're testing a build, so you should be taking a build manifest as input, not an input manifest.

Reply:
... or since you're taking a build ID, architecture and platform, construct the URL to the build manifest and fetch it from there